packages/crab-netting/src/CrabNetting.sol

```diff
@@ -7,49 +9,126 @@ contract CrabNetting {
+    mapping(address => uint256) public usd_balance;
+    mapping(address => uint256) public crab_balance;
     struct Receipt {
-        address depositor;
+        address sender;
         uint256 amount;
     }
-    Receipt[] deposits;
-    Receipt[] crab_deposits;
+    Receipt[] public deposits;
+    uint256 public depositsIndex;
+    Receipt[] public withdraws;
+    uint256 public withdrawsIndex;
 
     constructor(address _usdc, address _crab) {
         usdc = _usdc;
         crab = _crab;
     }
 
-    function depositUSDC(uint256 amount) public {
-        IERC20(usdc).transferFrom(msg.sender, address(this), amount);
-        usd_balance[msg.sender] = usd_balance[msg.sender] + amount;
-        deposits.push(Receipt(msg.sender, amount));
+    function depositUSDC(uint256 _amount) public {
+        IERC20(usdc).transferFrom(msg.sender, address(this), _amount);
+        usd_balance[msg.sender] = usd_balance[msg.sender] + _amount;
+        deposits.push(Receipt(msg.sender, _amount));
     }
 
-    function withdrawUSDC(uint256 amount) public {
-        require(usd_balance[msg.sender] >= amount);
-        usd_balance[msg.sender] = usd_balance[msg.sender] - amount;
-        IERC20(usdc).transfer(msg.sender, amount);
+    function withdrawUSDC(uint256 _amount) public {
+        // TODO ensure final version does not need this check
+        //require(usd_balance[msg.sender] >= _amount, "Withdrawing more than balance");
+        usd_balance[msg.sender] = usd_balance[msg.sender] - _amount;
+        IERC20(usdc).transfer(msg.sender, _amount);
     }
 
-    function depositCrab(uint256 amount) public {
-        IERC20(crab).transferFrom(msg.sender, address(this), amount);
-        crab_balance[msg.sender] = crab_balance[msg.sender] + amount;
-        crab_deposits.push(Receipt(msg.sender, amount));
+    function depositCrab(uint256 _amount) public {
+        IERC20(crab).transferFrom(msg.sender, address(this), _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] + _amount;
+        withdraws.push(Receipt(msg.sender, _amount));
     }
 
-    function withdrawCrab(uint256 amount) public {
-        require(crab_balance[msg.sender] >= amount);
-        crab_balance[msg.sender] = crab_balance[msg.sender] - amount;
-        IERC20(crab).transfer(msg.sender, amount);
+    function withdrawCrab(uint256 _amount) public {
+        // TODO ensure final version does not need this check
+        // require(crab_balance[msg.sender] >= _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] - _amount;
+        IERC20(crab).transfer(msg.sender, _amount);
     }
 
-    function balanceOf(address account) public view returns (uint256) {
-        return usd_balance[account];
+    function netAtPrice(uint256 _price, uint256 _quantity) public {
+        // TODO make only owner
+        uint256 forWithdraw = _quantity;
+        require(
+            _quantity <= IERC20(usdc).balanceOf(address(this)),
```

Comment:
We should make sure this one is 1e6, as in ur test u passing number with 18 decimals